src/Surfnet/ServiceProviderDashboard/Application/Dto/ChangeRequestDtoCollection.php

```diff
@@ -0,0 +1,46 @@
+<?php
+
+/**
+ * Copyright 2022 SURFnet B.V.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+namespace Surfnet\ServiceProviderDashboard\Application\Dto;
+
+use Webmozart\Assert\Assert;
+use Surfnet\ServiceProviderDashboard\Application\Dto\ChangeRequestDtoComparer;
+
+class ChangeRequestDtoCollection
+{
+    /**
+     * @var ChangeRequestDto[]
+     */
+    private $changeRequests = [];
+
+    public function __construct(array $changeRequests)
+    {
+        Assert::isArray($changeRequests);
+        Assert::isNonEmptyList($changeRequests, 'No change requests available');
```

Comment:
An empty collection should be allowed I guess. I ran into this Exception during my functional review. See feedback below for more info.